### ISSUE TITLE
Add thread menu item "create Filter from message"

### ISSUE THREAD

**Issue by RealRaven2000** [RealRaven2000/quickFilters#199]
Since Thunderbird 115 the context menu item "quickFilters: create filter from message" is missing. It would be helpful to get it back for direct access of this feature (rather than going through the toolbar menu).

![image](https://github.com/RealRaven2000/quickFilters/assets/6314674/977c8484-dda3-44f8-8035-a5ece921a8eb)

**Comment by RealRaven2000** [comment]
Here is a first version that supports creating filters from the context menu. At first it was a little tricky as it generated multiple assistant dialogs (if multiple windows are open), but I think I managed to filter out the wrong messages.

[quickFilters-wx-6.1pre15.zip](https://github.com/RealRaven2000/quickFilters/files/12134242/quickFilters-wx-6.1pre15.zip)

To install, download the zip file and drag into Thunderbird Add-ons Manager.

**Comment by RealRaven2000** [comment]
There were quite a few problems with retrieving the currently selected messages (using legacy methods), So I decided to rewrite the part of the code that determines these mails so that it is based on the API. Took quite a few tries but I think I managed this. So the selected mails are either one email that is right-clicked outside of the currently viewed mails (see picture below) or all currently viewed mails if the user clicks within those highlighted emails:

![image](https://github.com/RealRaven2000/quickFilters/assets/6314674/1d252fd4-37e1-42c5-ae54-6902128084a0)

Here is the latest prerelease: [quickFilters-wx-6.2pre64.zip](https://github.com/RealRaven2000/quickFilters/files/12329200/quickFilters-wx-6.2pre64.zip)

------

To test the version above, download the zip file, drag the file into Thunderbird Add-ons Manager, do not extract contents or it won't install.
